std/os.zig

```diff
@@ -1,495 +1,477 @@
-pub const windowsOpen = windows_util.windowsOpen;
-pub const windowsOpenW = windows_util.windowsOpenW;
-pub const createWindowsEnvBlock = windows_util.createWindowsEnvBlock;
-
-pub const WindowsCreateIoCompletionPortError = windows_util.WindowsCreateIoCompletionPortError;
-pub const windowsCreateIoCompletionPort = windows_util.windowsCreateIoCompletionPort;
-
-pub const WindowsPostQueuedCompletionStatusError = windows_util.WindowsPostQueuedCompletionStatusError;
-pub const windowsPostQueuedCompletionStatus = windows_util.windowsPostQueuedCompletionStatus;
-
-pub const WindowsWaitResult = windows_util.WindowsWaitResult;
-pub const windowsGetQueuedCompletionStatus = windows_util.windowsGetQueuedCompletionStatus;
-
-pub const WindowsWaitError = windows_util.WaitError;
-pub const WindowsOpenError = windows_util.OpenError;
-pub const WindowsWriteError = windows_util.WriteError;
-pub const WindowsReadError = windows_util.ReadError;
-
-pub const FileHandle = if (is_windows) windows.HANDLE else i32;
-
-pub const getAppDataDir = @import("os/get_app_data_dir.zig").getAppDataDir;
-pub const GetAppDataDirError = @import("os/get_app_data_dir.zig").GetAppDataDirError;
-
-const debug = std.debug;
-const assert = debug.assert;
-const testing = std.testing;
-
-const c = std.c;
-
-const mem = std.mem;
-const Allocator = mem.Allocator;
-
-const BufMap = std.BufMap;
-const cstr = std.cstr;
+pub use @import("os/bits.zig");
+
+/// See also `getenv`. Populated by startup code before main().
+pub var environ: [][*]u8 = undefined;
+
+/// Populated by startup code before main().
+/// Not available on Windows. See `std.process.args`
+/// for obtaining the process arguments.
+pub var argv: [][*]u8 = undefined;
+
+/// To obtain errno, call this function with the return value of the
+/// system function call. For some systems this will obtain the value directly
+/// from the return code; for others it will use a thread-local errno variable.
+/// Therefore, this function only returns a well-defined value when it is called
+/// directly after the system function call which one wants to learn the errno
+/// value of.
+pub const errno = system.getErrno;
+
+/// Closes the file descriptor.
+/// This function is not capable of returning any indication of failure. An
+/// application which wants to ensure writes have succeeded before closing
+/// must call `fsync` before `close`.
+/// Note: The Zig standard library does not support POSIX thread cancellation.
+pub fn close(fd: fd_t) void {
+    if (windows.is_the_target and !builtin.link_libc) {
+        return windows.CloseHandle(fd);
+    }
+    if (wasi.is_the_target) {
+        switch (wasi.fd_close(fd)) {
+            0 => return,
+            else => |err| return unexpectedErrno(err),
+        }
+    }
+    switch (errno(system.close(fd))) {
+        EBADF => unreachable, // Always a race condition.
+        EINTR => return, // This is still a success. See https://github.com/ziglang/zig/issues/2425
```

Comment:
Comment here isn't exactly correct. As mentioned in linked issue, some OSes you should retry, others your don't.

Also for OSX we should be calling `close_nocancel` here instead (that can be in another PR, but I'd leave a TODO here)

Reply:
I looked into close_nocancel via experimentation and looking for man pages on OSX and concluded that it's not a thing that exists anymore. Can you point to documentation for it?

Reply:
That's because it's called `close$NOCANCEL`, and it seems to be used at least by [Rust](https://github.com/rust-lang/libc/pull/596) and [Chrome](https://src.chromium.org/viewvc/chrome/trunk/src/base/mac/close_nocancel.cc?pathrev=223369).